modules/imgproc/test/test_drawing.cpp

```diff
@@ -680,6 +680,120 @@ TEST(Drawing, fillpoly_circle)
     EXPECT_LT(diff_fp3, 1.);
 }
 
+TEST(Drawing, fillpoly_contours)
+{
+    int imgSize = 50;
+    int type = CV_8UC1;
+    int shift = 0;
+    cv::LineTypes lineType = LINE_8;
```

Comment:
What about other line types? `LINE_AA`? Will it work the same as before?

Reply:
`LINE_AA` is not affected by the changes of this PR so it works the same, but it does not pass the test due to the reasons mentioned in the other comment.
`LINE_4` passes this test with any shift.
`LINE_8` passes this test only with a zero shift. It fails for non-zero shifts, most likely because the methods to draw the contour in `fillPoly` and `polylines` differ for `LINE_8` with non-zero shifts.